### DISCUSSION TITLE
Ampersands in XML

### DISCUSSION THREAD

**Discussion by davegraham92120** [mrwatson-de/fmWorkMate#39] [General]
I was using fmCheckMate to replace some embedded passwords in Import Records via ODBC script steps and found something odd. The new password contained ampersands, which caused the paste to FileMaker to fail. Ampersands must be encoded as `&amp;`. When you convert FM XML to text, any ampersands embedded in the schema come over encoded. I'm not sure if anything should be done or if it should just be left to documentation. Thoughts?

p.s. I guess it's not as odd as I thought since ampersands must be expressed as an entity reference in XML. https://www.w3.org/TR/xhtml1/guidelines.html#C_12

**Comment by mrwatson-de** [comment]
Hi @davegraham92120,

sorry, I missed your post.

Yep, as you say, a `&` symbol in FileMaker code will be encoded in XML text as `&amp;`

The problem is think clear, and that is

- fmWorkMate is a simple (i.e. dumb) search and replace utility
- You are working with XML text
- You'll have to encode special characters in the search and replace fields yourself

In other words, if you want to change your password 

- from: `mypassword`
- to: `my"Pass&Word"`

you'll actually have to enter the following text

- Find: `mypassword`
- Replace: `my&quot;Pass&amp;Word&quot;`

If you just replace it with `my"Pass&Word"`, yes, your XML will be broken!

Here are the characters you need to escape, if you are searching for / replacing them in XML text:

For|Use
---|---
 `"`|`&quot;`
`'`|`&apos;`
`<`|`&lt;`
`>`|`&gt;`
`&`|`&amp;`

Hope this helps

MrWatson 😄
